tsdb/store.go

```diff
@@ -804,6 +804,31 @@ func (s *Store) SetShardEnabled(shardID uint64, enabled bool) error {
 	return nil
 }
 
+// SetShardNewReadersBlocked sets if new readers can access the shard. If blocked
+// is true, the number of reader blocks is incremented and new readers will
+// receive an error instead of shard access. If blocked is false, the number
+// of reader blocks is decremented. If the reader blocks drops to 0, then
+// new readers will be granted access to the shard.
+func (s *Store) SetShardNewReadersBlocked(shardID uint64, blocked bool) error {
+	sh := s.Shard(shardID)
+	if sh == nil {
+		return ErrShardNotFound
```

Comment:
I'd like to wrap `ErrShardNotFound` with the shard ID and the blocked parameter, unless you are doing that high up the stack.

Reply:
That might help us catch something in a few years.

Reply:
All _current_ uses of `Store.SetShardNewReadersBlocked` and `Store.ShardInUse` will include the shard ID in the logging and the blocked parameter should be inferable from the log messages, but I'll go and ahead and this for future-proofing.

Reply:
Improved error message in latest commit.